File: integration-box/mqtt/ruby/mqtt.rb
require 'json'
require 'mqtt'

config = File.open(File.expand_path('../config.json', __dir__)) { |f|
  JSON.parse(f.read)
}

account_id = config['apikey'].split('/').first

client = MQTT::Client.new(config['broker'], :ssl => :TLSv1_2)
client.cert = File.read(config['certfile'])
client.key = File.read(config['keyfile'])
client.username = account_id
client.password = config['apikey']

client.connect('td-mqtt-sample')

target = "#{config['database']}:#{config['table']}"
topic = 'mqtt-ingest'
qos = 1

payload = {
    headers: {
      time: Time.now.to_i,
      auth: config['apikey'],
      target: target,
    },
    content: {
      id: 1,
      name: 'John Doe',
      age: 25,
      comment: 'hello, world',
      object: {lang: 'ruby'},
      array: %w[hello world]
    }
}
client.publish(topic, JSON.dump(payload), false, qos)

sleep 1

payload = {
    headers: {
      time: Time.now.to_i,
      auth: config['apikey'],
      target: target,
    },
    content: {
      id: 2,
      name: 'Jane Done',
      age: 23,
      comment: 'good morning',
      object: {lang: 'ruby'},
      array: %w[good morning]
    }
}
client.publish(topic, JSON.dump(payload), false, qos)

sleep 1

client.disconnect

File: integration-box/mqtt/python/mqtt.py
import json
import paho.mqtt.client as mqtt
from pathlib import Path
import ssl
import time


def on_log(client, userdata, level, buf):
    print('log: {}'.format(buf))


def on_disconnect(client, userdata, rc):
    client.loop_stop()


with Path(__file__).absolute().parent.parent.joinpath('config.json').open() as f:
    config = json.load(f)

account_id = config['apikey'].split('/')[0]

client = mqtt.Client()
client.on_log = on_log
client.on_disconnect = on_disconnect
client.tls_set(certfile=config['certfile'],
               keyfile=config['keyfile'],
               cert_reqs=ssl.CERT_NONE,
               tls_version=ssl.PROTOCOL_TLSv1_2)
client.username_pw_set(account_id, password=config['apikey'])

client.connect(config['broker'], 8883)
client.loop_start()

target = '{}:{}'.format(config['database'], config['table'])
topic = 'mqtt-ingest'
qos = 1

payload = {
    'headers': {
        'time': int(time.time()),
        'auth': config['apikey'],
        'target': target
    },
    'content': {
        'id': 1,
        'name': 'John Doe',
        'age': 25,
        'comment': 'hello, world',
        'object': {'lang': 'python'},
        'array': ['hello', 'world']
    }
}
client.publish(topic, json.dumps(payload), qos)

time.sleep(1)

payload = {
    'headers': {
        'time': int(time.time()),
        'auth': config['apikey'],
        'target': target
    },
    'content': {
        'id': 2,
        'name': 'Jane Doe',
        'age': 23,
        'comment': 'good morning',
        'object': {'lang': 'python'},
        'array': ['good', 'morning']
    }
}
client.publish(topic, json.dumps(payload), qos)

time.sleep(1)

client.disconnect()
